src/components/carousel/carousel.ts

```diff
@@ -0,0 +1,769 @@
+   */
+  public async select(
+    slide: IgcCarouselSlideComponent,
+    direction?: 'next' | 'prev'
+  ): Promise<boolean> {
+    const index = this.slides.indexOf(slide);
+
+    if (index === this.current || index === -1) {
+      return false;
+    }
+
+    const dir = direction ?? (index > this.current ? 'next' : 'prev');
+
+    await this.animateSlides(slide, this._activeSlide, dir);
+    return true;
+  }
+
+  private navigationTemplate() {
+    return html`
+      <igc-button
+        ${ref(this._prevButtonRef)}
+        type="button"
+        part="navigation previous"
+        aria-label="Previous slide"
+        aria-controls=${this._carouselId}
+        ?disabled=${this.skipLoop && this.current === 0}
+        @click=${() => this.handleNavigationInteraction('prev')}
+      >
+        <slot name="previous-button">
+          <igc-icon
+            name="carousel_prev"
+            collection="default"
+            aria-hidden="true"
+          ></igc-icon>
+        </slot>
+      </igc-button>
+      <igc-button
+        ${ref(this._nextButtonRef)}
+        type="button"
+        part="navigation next"
+        aria-label="Next slide"
+        aria-controls=${this._carouselId}
+        ?disabled=${this.skipLoop && this.current === this.total - 1}
+        @click=${() => this.handleNavigationInteraction('next')}
+      >
+        <slot name="next-button">
+          <igc-icon
+            name="carousel_next"
+            collection="default"
+            aria-hidden="true"
+          ></igc-icon>
+        </slot>
+      </igc-button>
+    `;
+  }
+
+  protected *renderIndicators() {
+    for (let i = 0; i < this.slides.length; i++) {
+      const slide = this.slides[i];
+      const forward = `${slide.active ? 'visible' : 'hidden'}`;
+      const backward = `${slide.active ? 'hidden' : 'visible'}`;
+      yield html`<igc-carousel-indicator
+        exportparts="indicator, active, inactive"
+        role="tab"
+        tabindex=${slide.active ? 0 : -1}
+        aria-label="Slide ${i + 1}"
+        aria-selected=${slide.active}
+      >
+        <div
+          part="dot"
+          style=${styleMap({ visibility: backward, zIndex: 1 })}
+        ></div>
+        <div
+          part="dot active"
+          slot="active"
+          style=${styleMap({ visibility: forward })}
+        ></div>
+      </igc-carousel-indicator>`;
+    }
+  }
+
+  private indicatorTemplate() {
+    return html`
+      <igc-carousel-indicator-container>
+        <div
+          ${ref(this._indicatorsContainerRef)}
+          role="tablist"
+          part=${partNameMap({
+            indicators: true,
+            start: this.indicatorsOrientation === 'start',
+          })}
+        >
+          <slot
+            name="indicator"
+            @slotchange=${this.handleIndicatorSlotChange}
+            @click=${this.handleIndicatorClick}
+          >
+            ${this.renderIndicators()}
```

Comment:
```suggestion
      ${this.hasProjectedIndicators
        ? this.renderProjectedIndicators() <- better name would be updateProjectedIndicators
        : this.renderIndicators()}
```

where renderProjectedIndicators would be along the lines of (obviously not checked for bugs):
```ts
protected renderProjectedIndicators() {
    for (const [idx, slide] of this.slides.entries()) {
      const indicator = this._projectedIndicators[idx];
      indicator.role = 'tab'; // <- this can be set as a default ElementInternals role for the indicator component
      indicator.tabIndex = slide.active ? 0 : -1;
      indicator.ariaLabel = `Slide ${idx + 1}`;
      indicator.ariaSelected = slide.active ? 'true' : 'false';
      indicator.setAttribute('aria-controls', slide.id);
      this.setIndicatorVisibility(indicator, slide.active); // <- after refactor can be only indicator.active = slide.active;
    }
  }
```

This is way easier to track as a state since it is on the render "hook" and would trigger correctly on each reactive update of the carousel.

Reply:
Sure, I guess it is still the issue with the axe testing tools which don't really account for ElementInternals when they do run the checks. Good enough as is currently.